lib/http/request/body.rb

```diff
@@ -32,14 +29,10 @@ def size
       #
       # @yieldparam [String]
       def each
-        return enum_for(__method__) unless block_given?
-
         if @body.is_a?(String)
           yield @body
         elsif @body.respond_to?(:read)
-          while (data = @body.read(BUFFER_SIZE))
-            yield data
-          end
+          IO.copy_stream(@body, BlockIO.new(Proc.new))
```

Comment:
IMHO it worth to simply change `each` signature to `def each(&block)` and pass the block here, and in next clause (`@body.each(&block)`)